### DISCUSSION TITLE
what is the default delivery technique?

### DISCUSSION THREAD

**Discussion by junjunjanna** [e0404/matRad#666] [Q&A]
hello everyone, 
I'm a little confused about the delivery technique for proton and carbon-ion.
 when I use the Generic machine data, is that delivered by passive scattering or active scanning during the planning process?

**Comment by wahln** [comment]
I suggest to read the [paper](https://aapm.onlinelibrary.wiley.com/doi/abs/10.1002/mp.12251) and our [Wiki](https://github.com/e0404/matRad/wiki).

matRad is built for intensity modulated planning, and thus the Generic data and the dose calculation algorithms are tuned for scanning. It is possible to include a passive scattering beam through some hacking of how the pencil-beam kernels in the database are convolved with the incoming fluence (if this is known after the passive beam elements).

**Reply by junjunjanna** [reply]
Thanks for your reply, wahln!
If that is a scanning system, would that be possible to show the position of the Bragg peak of each scanning spot (as image attached  below)?
<img width="395" alt="Screenshot 2023-10-10 at 15 56 54" src="https://github.com/e0404/matRad/assets/114455710/647cc51c-30a5-413b-8c00-8dfe5235b428">

**Reply by wahln** [reply]
There's unfortunately no built-in function to dipslay that. The spot positions / the spot grid can, however, obtained from the `stf` within matRad. The lateral positions in the isocenter are stored within `stf.ray` under the variable `rayPos`, and each of these "rays" has mutliple energies from which the position in water-equivalent depth can be obtained. From this one could (using the ray-tracing algorithm) calculate each spot position in world coordinates and then display them.

**Reply by junjunjanna** [reply]
I see, thank you so much, wahln! 
Just one more question related, is matRad using any algorithm to optimize the scanning path during particle treatment planning?

**Reply by wahln** [reply]
There is no dedicated algorithm to optimize the scan path. Would be excited to have one in matRad though, in case you want to develop one ;-)
